src/mono/mono/metadata/appdomain.c

```diff
@@ -1239,15 +1249,25 @@ void
 mono_runtime_register_appctx_properties (int nprops, const char **keys,  const char **values)
 {
 	n_appctx_props = nprops;
-	appctx_keys = g_new0 (gunichar2*, nprops);
-	appctx_values = g_new0 (gunichar2*, nprops);
+	appctx_keys = g_new0 (char*, n_appctx_props);
+	appctx_values = g_new0 (char*, n_appctx_props);
 
 	for (int i = 0; i < nprops; ++i) {
-		appctx_keys [i] = g_utf8_to_utf16 (keys [i], strlen (keys [i]), NULL, NULL, NULL);
-		appctx_values [i] = g_utf8_to_utf16 (values [i], strlen (values [i]), NULL, NULL, NULL);
+		appctx_keys [i] = g_new0 (char, strlen (keys [i]) + 1);
+		appctx_values [i] = g_new0 (char, strlen (values [i]) + 1);
+		strcpy(appctx_keys [i], keys [i]);
+		strcpy(appctx_values [i], values [i]);
```

Comment:
```suggestion
		strcpy (appctx_keys [i], keys [i]);
		strcpy (appctx_values [i], values [i]);
```

Reply:
wouldn't g_strdup be easier and safer here?

Reply:
Fixed. Thanks